### ISSUE TITLE
codepipeline: DescribeChangeSet permissions too restrictive for PipelineCreateReplaceChangeSetAction

### ISSUE THREAD

**Issue by clareliguori** [awslabs/aws-cdk#1034]
It appears that the cloudformation:DescribeChangeSet permission automatically added to the pipeline role by the PipelineCreateReplaceChangeSetAction class is too restrictive, and should not add the "cloudformation:ChangeSetName" condition.

https://github.com/awslabs/aws-cdk/blob/5bb7c93f283274263fb77445f8ffd529c37cd74b/packages/%40aws-cdk/aws-cloudformation/lib/pipeline-actions.ts#L269-L273

The first time through the pipeline (creating a change set for a new CFN stack), the action results in the following error:
`User: arn:aws:sts::xxxxx:assumed-role/Pipeline-PipelineRoleB27FAA37-FJET4A98XNMY/1540783303789 is not authorized to perform: cloudformation:DescribeChangeSet on resource: arn:aws:cloudformation:us-east-2:xxxxx:stack/infrastructure-test/*`

CDK version: 0.14.1

CDK code snippet:
```
new cfn.PipelineCreateReplaceChangeSetAction(this, 'PrepareChangesTest', {
            stage: testStage,
            stackName: testStackName,
            changeSetName,
            runOrder: 1,
            fullPermissions: true,
            templatePath: buildAction.outputArtifact.atPath(templatePrefix + 'Test.template.yaml'),
        });
```

Policy snippet generated by CDK:
```
        {
            "Condition": {
                "StringEquals": {
                    "cloudformation:ChangeSetName": "StagedChangeSet"
                }
            },
            "Action": [
                "cloudformation:CreateChangeSet",
                "cloudformation:DeleteChangeSet",
                "cloudformation:DescribeChangeSet"
            ],
            "Resource": "arn:aws:cloudformation:us-east-2:xxxxx:stack/infrastructure-test/*",
            "Effect": "Allow"
        }
```

Adding the following policy to the pipeline role makes the action successfully complete:
```
{
            "Action": [
                "cloudformation:DescribeChangeSet"
            ],
            "Resource": "arn:aws:cloudformation:us-east-2:xxxxx:stack/infrastructure-test/*",
            "Effect": "Allow"
        }
```

**Comment by RomainMuller** [comment]
The problem is likely fixed by a recent PR already, I'll double check that & relate if that is the case.

**Comment by RomainMuller** [comment]
Yah, I reckon the issue was fixed as a side-effect of #1009.

**Comment by rix0rrr** [comment]
Closing because of no response.
